sidechain/consensus/slots/src/slot_stream.rs

```diff
@@ -0,0 +1,106 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Implementation of schedule of the slots to time so that trusted calls can be aligned.
+
+use crate::time_until_next_slot;
+use futures_timer::Delay;
+use std::time::Duration;
+
+/// Triggers the enclave to produce a block based on a fixed time schedule.
+pub async fn start_interval_block_production<F>(trusted_call: F, slot_duration: Duration)
+where
+	F: Fn(),
+{
+	let mut slot_stream = SlotStream::new(slot_duration);
+
+	loop {
+		slot_stream.next_slot().await;
+		trusted_call();
+	}
+}
+
```

Comment:
I would name this function more generically.

`start_slot_worker<F>(task: F, slot_duration)`;

Accordingly, we can make the documentation more generic:

```
Executes given `task` repeatedly when the next slot becomes available.
```